src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -1786,72 +1788,95 @@ private ScheduleResponse TrySatisfyRequestFromCache(int nodeForResults, BuildReq
-                BuildResult result = new BuildResult(request, new InvalidOperationException(errorMessage));
-                result.SetOverallResult(false);
+            // Issue a failed build result to have the msbuild task marked as failed and thus stop the build
+            BuildResult result = new BuildResult(request, new InvalidOperationException(errorMessage));
+            result.SetOverallResult(false);
 
-                var response = GetResponseForResult(nodeForResults, request, result);
-                responses.Add(response);
+            var response = GetResponseForResult(nodeForResults, request, result);
+            responses.Add(response);
 
-                // Log an error to have something displayed to the user and to avoid having a failed build with 0 errors
-                // todo Search if there's a way to have the error automagically logged in response to the failed build result
-                _componentHost.LoggingService.LogErrorFromText(
-                    new BuildEventContext(
-                        request.SubmissionId,
-                        1,
-                        BuildEventContext.InvalidProjectInstanceId,
-                        BuildEventContext.InvalidProjectContextId,
-                        BuildEventContext.InvalidTargetId,
-                        BuildEventContext.InvalidTaskId),
-                    null,
-                    null,
-                    null,
-                    new BuildEventFileInfo(requestConfig.ProjectFullPath),
-                    errorMessage);
+            // Log an error to have something displayed to the user and to avoid having a failed build with 0 errors
+            // todo Search if there's a way to have the error automagically logged in response to the failed build result
+            _componentHost.LoggingService.LogErrorFromText(
+                NewBuildEventContext(),
+                null,
+                null,
+                null,
+                new BuildEventFileInfo(requestConfig.ProjectFullPath),
+                errorMessage);
 
-                return false;
+            return false;
+
+            BuildEventContext NewBuildEventContext()
+            {
+                return new BuildEventContext(
+                    request.SubmissionId,
+                    1,
+                    BuildEventContext.InvalidProjectInstanceId,
+                    BuildEventContext.InvalidProjectContextId,
+                    BuildEventContext.InvalidTargetId,
+                    BuildEventContext.InvalidTaskId);
             }
 
-            return true;
+            (BuildRequestConfiguration requestConfig, BuildRequestConfiguration parentConfig) GetConfigurations()
+            {
+                var buildRequestConfiguration = configCache[request.ConfigurationId];
+
+                // Need the parent request. But the parent / child relationship is sometimes formed after the cache is queried, so get creative
```

Comment:
```suggestion
                // Need the parent request. But the parent / child relationship is sometimes formed after the cache is populated, so get creative
```

? Also, if it's "sometimes" should there be an optimistic check?

Reply:
What's an optimistic check?

I check in two places because I don't quite understand if it would be safe enough to just check in one of them. From what I've noticed, a request migrates back and forth between BlockedRequests and ExecutingRequests, and it seems that the parent could be in either when the cache miss constraint is checked.

Reply:
Ah, I see. Let's try a small rewrite:

```suggestion
                // Need the parent request. It might be blocked or executing; check both
```

?